packages/puppeteer-core/src/common/bidi/Frame.ts

```diff
@@ -54,13 +54,14 @@ export class Frame extends BaseFrame {
     this._id = this.#context.id;
     this._parentId = parentId ?? undefined;
 
+    const puppeteerRealm = context.createSandboxRealm(UTILITY_WORLD_NAME);
     this.sandboxes = {
       [MAIN_SANDBOX]: new Sandbox(context, timeoutSettings),
-      [PUPPETEER_SANDBOX]: new Sandbox(
-        context.createSandboxRealm(UTILITY_WORLD_NAME),
-        timeoutSettings
-      ),
+      [PUPPETEER_SANDBOX]: new Sandbox(puppeteerRealm, timeoutSettings),
     };
+
+    puppeteerRealm.setFrame(this);
+    context.setFrame(this);
```

Comment:
Why is this change need?

Reply:
Sometimes inputs are performed in the puppeteer realm without transferring to the main sandbox. This causes issues because we need a reference to the page which is obtained through the frame on the context if it exists.